plugin/src/generators/lib-rename/generator.ts

```diff
@@ -0,0 +1,45 @@
+import { Tree } from '@nx/devkit';
+import { LibRenameGeneratorSchema } from './schema';
+import { askQuestion, getLibrariesDetails, selectLibrary } from '../../shared/utils';
+import { execSync } from 'child_process';
+
+export async function libRenameGenerator(
+  tree: Tree,
+  options: LibRenameGeneratorSchema
+) {
+  let srcLibraryName: string;
+  let srcLibraryPath: string;
+
+  if (options.srcLibName) {
+    srcLibraryName = options.srcLibName;
+
+    const library = getLibrariesDetails(tree).find((library) => library.name === srcLibraryName);
+
+    if (!library) {
+      throw new Error(`Library ${srcLibraryName} not found`);
+    }
+
+    srcLibraryPath = library.path;
+  } else {
+    const selectedLibrary = await selectLibrary(tree, 'Select the library to rename: ');
+
+    srcLibraryName = selectedLibrary.name;
+    srcLibraryPath = selectedLibrary.path;
+  }
+  
+  const libPathSegments = srcLibraryPath.split('/');
+  const defaultDestLibraryName = libPathSegments.pop();
```

Comment:
What if a library name, excluding app, scope and type, has 2+ words?

Reply:
I can't understand what do you mean. This code takes original name of the lib (e.g.: `mobile-account-features-profile-settings` -> `profile-settings` but using library path: `libs/mobile/account/features/profile-settings` -> `profile-settings`).

Reply:
Sorry, I was confused about the signs for separation 🙃 You are right